File: 1.two-sum.java
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
// import java.util.*;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] res = new int[2];

        for (int i=0;i<nums.length;i++){
            if (!map.containsKey(target-nums[i])) map.put(nums[i], i);
            else {
                res[1] = i;
                res[0] = map.get(target-nums[i]);
                return res;
            }
        }
        return null;

    }
}
// @lc code=end

File: 1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;

        for (int i = 0; i < nums.size(); i++) {
            const int n = nums[i];
            const int newTarget = target - n;
            
            if (map.find(newTarget) != map.end()) 
                return {map[newTarget], i};
            map[n] = i;
        }
        return {};
    }
};
// @lc code=end

